### PR TITLE
ogr2ogr: -emptyStrAsNull option

### PR DESCRIPTION
**Pull request by bjornharrtell**
Follow up to https://github.com/OSGeo/gdal/pull/2209.

Testcase verified working locally.

### CODE REVIEW

File: autotest/utilities/test_ogr2ogr_lib.py [hunk 1]
```diff
@@ -603,3 +602,24 @@ def test_ogr2ogr_lib_sql_filename():
...
+    f = ogr.Feature(lyr.GetLayerDefn())
+    f['foo'] = ''
+    lyr.CreateFeature(f)
+
+    ds = gdal.VectorTranslate('', src_ds, format='Memory', emptyStrAsNull=True)
+
+    lyr = ds.GetLayer(0)
+    f = lyr.GetNextFeature()
+
+    assert f['foo'] == None, 'expected empty string to be transformed to null'
```
Comment (rouault) [inline autotest/utilities/test_ogr2ogr_lib.py:624]: Probably slightly more idiomatic:
```suggestion
    assert f['foo'] is None, 'expected empty string to be transformed to null'
```

**Review by rouault — COMMENTED** [review]
You need to commit the regenerated Python bindings file as well (with SWIG 3.0.12 to avoid churn)

**Comment by bjornharrtell** [comment]
@rouault bindings added (wasn't easy!)

I don't see anything explaining the remaining ci failure.

**Comment by rouault** [comment]
> I don't see anything explaining the remaining ci failure.

Hum, we have had similar mysterious issue in the past. Can you try to add a commit with the following patch ?

```
diff --git a/.github/workflows/ubuntu_18.04/start.sh b/.github/workflows/ubuntu_18.04/start.sh
index 6d8f3fc..d17c5c8 100755
--- a/.github/workflows/ubuntu_18.04/start.sh
+++ b/.github/workflows/ubuntu_18.04/start.sh
@@ -9,6 +9,9 @@ DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
 TRAVIS=yes
 export TRAVIS
 
+BUILD_NAME=ubuntu_1804
+export BUILD_NAME
+
 LANG=en_US.UTF-8
 export LANG
 apt-get install -y locales && \
diff --git a/autotest/gdrivers/bag.py b/autotest/gdrivers/bag.py
index b2a5825..46cbfb4 100755
--- a/autotest/gdrivers/bag.py
+++ b/autotest/gdrivers/bag.py
@@ -51,8 +51,8 @@ def check_no_file_leaks():
 
     diff = len(gdaltest.get_opened_files()) - num_files
 
-    if diff != 0 and gdaltest.is_travis_branch('trusty_clang'):
-        print('Mysterious leak of file handle on trusty_clang')
+    if diff != 0 and gdaltest.is_travis_branch('ubuntu_1804'):
+        print('Mysterious leak of file handle on ubuntu_1804')
         return
 
     assert diff == 0, 'Leak of file handles: %d leaked' % diff
```

**Comment by bjornharrtell** [comment]
@rouault patch added but all green now 🤷

**Comment by rouault** [comment]
Could you rework the history to have 1 commit with the actual work, and another one with the CI patch (with a commit message like "CI: skip test on ubuntu_1804" for the person who will compile commit logs to create release notes) ?

**Comment by bjornharrtell** [comment]
yep no problem, done and force pushed.

**Pull request merged**